### PR TITLE
Add tests for `LocalCommunity`

### LINKED ISSUE #1453: Missing unit tests for `LocalCommunity`
`LocalCommunity` currently appears to have no dedicated unit tests.
It is used by SelectiveCommunityDetector algorithms.

### PR DESCRIPTION
**Pull request by Schwarf**
This PR adds unit tests for the class template `LocalCommunity`.
The new tests cover the explicit template instantiations, including add/remove behavior, shell bookkeeping, cut/internal weight updates and boundary tracking.

While adding the tests, this also fixes a boundary bookkeeping bug. 

Fixes #1453

### CODE REVIEW

File: networkit/cpp/structures/test/LocalCommunityGTest.cpp [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+auto collectCommunity(LocalCommunity &community, F getValue) {
+    using Value =
+        decltype(getValue(std::declval<const typename LocalCommunity::CommunityInfo &>()));
+    std::unordered_map<node, Value> result;
+    community.forCommunityNodes(
+        [&](node u, const auto &info) { result.emplace(u, getValue(info)); });
+    return result;
+}
+
+Graph weightedGraph() {
```
Comment (angriman) [inline networkit/cpp/structures/test/LocalCommunityGTest.cpp:66]: Nit: As already done in other tests, consider adding a comment with a drawing of how the graph looks like.
```cpp
//        (5)
//         |  13
//        (0)
//         |  2
//        (1)------ 3 ------(2)
//         |                 |
//         5                 7
//         |                 |
//         +------(3)--------+
//                 |  11
//                (4)
```

File: networkit/cpp/structures/test/LocalCommunityGTest.cpp [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    }
+
+    if constexpr (TypeParam::maintainBoundary) {
+        EXPECT_EQ(community.boundarySize(), 1u);
+    }
+
+    community.addNode(2);
+    community.addNode(3);
+
+    EXPECT_EQ(community.toSet(), std::set<node>({1, 2, 3}));
```
Comment (angriman) [inline networkit/cpp/structures/test/LocalCommunityGTest.cpp]: Nit: Prefer using `ElementsAre`, same in tests below.

File: networkit/cpp/structures/test/LocalCommunityGTest.cpp [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+using RemovableLocalCommunityConfigs = ::testing::Types<LocalCommunityConfig<false, false, true>,
+                                                        LocalCommunityConfig<true, false, true>,
+                                                        LocalCommunityConfig<true, true, true>>;
+
+TYPED_TEST_SUITE(RemovableLocalCommunityGTest, RemovableLocalCommunityConfigs,
+                 /*Comma needed for variadic macro.*/);
+
+TYPED_TEST(RemovableLocalCommunityGTest, testRemoveNodeMaintainsCommunityShellAndCut) {
+    Graph G = weightedGraph();
```
Comment (angriman) [inline networkit/cpp/structures/test/LocalCommunityGTest.cpp]: Nit: This can be `const`.
```suggestion
    const Graph G = weightedGraph();
```

File: networkit/cpp/structures/test/LocalCommunityGTest.cpp (hunk 1 above)
Comment (Schwarf) [reply networkit/cpp/structures/test/LocalCommunityGTest.cpp:66]: Done

File: networkit/cpp/structures/test/LocalCommunityGTest.cpp (hunk 2 above)
Comment (Schwarf) [reply networkit/cpp/structures/test/LocalCommunityGTest.cpp]: Done

File: networkit/cpp/structures/test/LocalCommunityGTest.cpp (hunk 3 above)
Comment (Schwarf) [reply networkit/cpp/structures/test/LocalCommunityGTest.cpp]: Done

**Review by angriman — APPROVED** [review]
